clients/rust/src/lib.rs

```diff
@@ -68,18 +68,18 @@ impl SysadminClient {
     /// "address" will become a SocketAddr using the trait
     /// `[ToSocketAddrs]`: https://doc.rust-lang.org/std/net/trait.ToSocketAddrs.html
     pub fn connect<A: ToSocketAddrs>(&mut self, address: A) -> Result<()> {
-        let stream = TcpStream::connect(address).chain_err(|| {
-            ErrorKind::SysadminConnectionError("Could not connect to Sysadmin".to_string())
+        let stream = TcpStream::connect(address).or_else(|_err| {
```

Comment:
I think a better way to attach contextual information to things like you've done here is to use the `context` method on `failure::Error`. Something like:

```rust
let stream = TcpStream::connect(address).context("failed to connect to sysadmin")?;
```

Your custom defined `SysadminErrorKind` should only really be used for errors that originate from this crate (and not errors that you simply want to bubble upwards to your caller).